File: scripts/threejs10x.mjs
import 'zx/globals'

for (const i in Array(10).fill(null)) {
  fs.ensureDir('./temp/threejs10x')
  await $`cp -r ./temp/threejs/src ./temp/threejs10x/copy${i}`
}

const entryCode = Array(10)
  .fill(null)
  .map((_, i) => i)
  .map(
    (i) =>
      `import * as copy${i} from './copy${i}/Three.js'\nexport { copy${i} }`,
  )
  .join('\n')

fs.writeFile('./temp/threejs10x/main.js', entryCode)

File: crates/rolldown/examples/threejs10x.rs
use std::path::PathBuf;

use rolldown::{Bundler, InputItem, InputOptions, OutputOptions};
use rolldown_tracing::enable_tracing_on_demand;
use sugar_path::SugarPathBuf;

#[tokio::main]
async fn main() {
  let _guard = enable_tracing_on_demand();
  let project_root = std::env::var_os("CARGO_MANIFEST_DIR")
    .map(|s| PathBuf::from(s))
    .unwrap_or_else(|| {
      let project_root = std::env::current_dir()
        .unwrap()
        .join("./crates/rolldown")
        .into_normalize();
      project_root
    });

  let mut bundler = Bundler::new(InputOptions {
    input: vec![InputItem {
      name: "threejs-10x".to_string(),
      import: project_root
        .join("../../temp/threejs10x/main.js")
        // .join("../../temp/threejs/src/Three.js")
        .to_string_lossy()
        .to_string(),
    }],
    cwd: project_root.join("../../temp/threejs10x/"),
    // .join("../../temp/threejs"),
    ..Default::default()
  });
  bundler
    .write(OutputOptions {
      ..Default::default()
    })
    .await
    .unwrap();
}
